src/ports.rs

```diff
@@ -435,9 +440,9 @@ struct PortCell<T: Sync> {
     /// - if you then try binding C -> A, then we can know
     /// that C is in the downstream of A, indicating that there is a cycle.
     #[cfg(feature = "no-unsafe")]
-    downstreams: AtomicRefCell<HashMap<PortId, Rc<AtomicRefCell<Rc<PortCell<T>>>>>>,
+    downstreams: DownstreamsSafe<T>,
```

Comment:
Looks like if you're using a conditional alias you can also use a single non-conditional field?